ciphers/caesar/caesar_test.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"testing"
+)
+
+var caesarTestData = []struct {
+	description string
+	input       string
+	key         int
+	expected    string
+}{
+	{
+		"Basic caesar encryption with letter 'a'",
+		"a",
+		3,
+		"d",
+	},
+	{
+		"Basic caesar encryption wrap around alphabet on letter 'z'",
+		"z",
+		3,
+		"c",
+	},
+	{
+		"Encrypt a simple string with ceasar encryiption",
+		"hello",
+		3,
+		"khoor",
+	},
+	{
+		"Encrypt a simple string with key 13",
+		"hello",
+		13,
+		"uryyb",
+	},
+	{
+		"Encrypt a simple string with key -13",
+		"hello",
+		-13,
+		"uryyb",
+	},
+	{
+		"With key of 26 output should be the same as the input",
+		"no change",
+		26,
+		"no change",
+	},
+	{
+		"Encrpyt sentence with key 10",
+		"the quick brown fox jumps over the lazy dog.",
+		13,
+		"gur dhvpx oebja sbk whzcf bire gur ynml qbt.",
+	},
+}
+
+func TestCeasarCipher(t *testing.T) {
+	for _, test := range caesarTestData {
+		actual := caesarCipher(test.input, test.key)
```

Comment:
I would suggest to run each iteration in a t.Run() func. This would ensure that even one of the test cases fail, rest of them will still get executed. Currently, if a test case fails, the test terminates and does not execute the remaining cases.

Reply:
Yes, I agree that for now it will fail at the first failing test case. I'm changing it so that it executes the rest. Thanks for your feedback!